addon/utils/function-modifier.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {
+  capabilities as modifierCapabilities,
+  setModifierManager,
+} from '@ember/modifier';
+
+type LazyTrackedArgs = {
+  positional?: Array<unknown>;
+  named?: Record<string, unknown>;
+};
+
+type FunctionModifier<Args extends LazyTrackedArgs> =
+  | ((
+      element: HTMLElement,
+      args: Args['positional'],
+      namedArgs?: Record<string, any>
+    ) => () => void)
+  | ((element: HTMLElement, args: Args['positional']) => void);
+
+interface ModifierState {
+  fn: FunctionModifier<LazyTrackedArgs>;
+  args: LazyTrackedArgs;
+  // normally this isn't a HTMLElement guarantee, but since this modifier manager has 1 use,
+  // we can make the assumption
+  element?: HTMLElement;
+  destructor?: void | (() => void);
+}
+
+/**
+ *  This is copied from
+ *  https://github.com/pzuraq/ember-could-get-used-to-this/blob/master/addon/-private/modifiers.js
+ *
+ * Because:
+ *  - we don't want to *force* folks into use ember-could-get-used-to-this
```

Comment:
No against including this here, just wondering why we don't want to depend on `ember-could-get-used-to-this`?

Reply:
we absolutely could! what other things might we want to use `ember-could-get-used-to-this` for?
(I think we need to move ember-could-get-used-to-this in to adopted-addons, or specifically extract out the function-modifier bit to a new addon and have that included everywhere -- maybe call it `ember-default-modifier-manager-polyfill` or something related to the matching RFC 🤷)